api/src/main/java/org/eclipse/microprofile/rest/client/TypeSafeClientBuilder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2017 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.rest.client;
+
+import javax.annotation.Priority;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.ServiceLoader;
+
+public abstract class TypeSafeClientBuilder {
+    public static TypeSafeClientBuilder newBuilder() {
+        ServiceLoader<TypeSafeClientBuilder> loader = ServiceLoader.load(TypeSafeClientBuilder.class);
+        List<TypeSafeClientBuilder> clientBuilders = new ArrayList<>();
+        loader.forEach(clientBuilders::add);
+        loader = ServiceLoader.load(TypeSafeClientBuilder.class, TypeSafeClientBuilder.class.getClassLoader());
+        loader.forEach(clientBuilders::add);
+
+        if(clientBuilders.size() == 0) {
+            throw new RuntimeException("No implementation of '"+TypeSafeClientBuilder.class.getSimpleName()+"' found");
+        }
+        clientBuilders.sort(Comparator.comparingInt(value -> {
+            Priority priority = value.getClass().getAnnotation(Priority.class);
+            if (priority == null) {
+                return 1;
+            }
+            else {
+                return priority.value();
+            }
+        }).reversed());
```

Comment:
Is the intent that higher value in the @Priority wins? If so, that is inconsistent with other uses of @Priority, particularly JAX-RS where higher priority actually has a lower value - i.e. @Priority(1) wins over @Priority(2).

Reply:
In the case of JAX-RS, multiple filters will apply.  The use of the highest priority wins is consistent with the CDI spec, when choosing observer method order, alternatives, etc.  So yes, the intent is that the highest priority impl will win (granted this is just to handle the case when multiple are on the classpath for some odd reason)